### PR TITLE
feat(tabs): paginated tab list

### LINKED ISSUE #203: Paginated tabs
### Which scope/s are relevant/related to the feature request?

tabs

### Information

Providing a way to scroll through multiple tabs with paginated header. This would be helpful when displaying multiple tabs or for mobile view.

Example from Angular Material Tabs: https://material.angular.io/components/tabs/examples#tab-group-paginated

The paginated header should show before/after buttons when the header needs to scroll, otherwise hides the buttons.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [X] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by marcjulian**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] input
- [ ] label
- [ ] menubar
- [ ] navigation-menu
- [ ] pagination
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] spinner
- [ ] switch
- [ ] table
- [x] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

Closes #203

## What is the new behavior?

Add `hlm-paginated-tab-list` component which handles pagination built on top of [MatPaginatedTabHeader](https://github.com/angular/components/blob/main/src/material/tabs/paginated-tab-header.ts)

CleanShot 2024-03-14 at 11 23 18

Current issues I am facing:

1. Doesn't scroll on initial load to another tab
Should find the position of the "Tab 20" and scroll it into view.
CleanShot 2024-03-14 at 11 29 54

3. Scroll position is calculated from tabs trigger, but the extra padding (`p-1`) on the list interferes with the correct scroll position
When scrolling left with the keyboard it scrolls to "Tab 29" but does not consider the extra padding. The next paginator button is also still enabled, but should be disabled when it is at the end.
CleanShot 2024-03-14 at 11 31 00
This is when it is scrolled all the way to the end. The next paginator is disabled and the padding around the tab is visible.
CleanShot 2024-03-14 at 11 32 50

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/ui/tabs/brain/src/lib/brn-tabs-paginated-list.directive.ts [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+		if (currentKey && this._items) {
+			// TODO currentTab is undefined for initial load
+			const currentTab = this._tabs[currentKey];
+			if (currentTab) {
+				activeIndex = this._items.toArray().indexOf(currentTab.trigger);
+			}
+		}
+
+		return activeIndex;
+	});
```
Comment (marcjulian) [inline libs/ui/tabs/brain/src/lib/brn-tabs-paginated-list.directive.ts:170]: I am calculating the `selectedIndex` based on the roots tab value. Then I need to get the current tab for the tab value to calculate the position of the tab. However, the `currentTab` is always `undefined` on initial load. Probably the tab value is available before the `tabs` have been registered.

Comment (goetzrobin) [reply libs/ui/tabs/brain/src/lib/brn-tabs-paginated-list.directive.ts:170]: I think tabs was created pre signals. What if we make `tabs` a signal?

Comment (marcjulian) [reply libs/ui/tabs/brain/src/lib/brn-tabs-paginated-list.directive.ts]: Yes that did the trick! The initial tab is now correctly calculated and scrolled into view.

**Comment by marcjulian** [comment]
@goetzrobin do you have an idea how to handle the extra padding around the tabs? I noticed when I add padding to the material tabs it behaves the same as with the current solution. 

Not sure if we need to specify the padding amount to consider while scrolling.

**Comment by goetzrobin** [comment]
> @goetzrobin do you have an idea how to handle the extra padding around the tabs? I noticed when I add padding to the material tabs it behaves the same as with the current solution.
> 
> Not sure if we need to specify the padding amount to consider while scrolling.

I think this is a tough one to figure out. I'd say as good as Angular Material is good enough for me. What do you think?

**Comment by marcjulian** [comment]
Yes think so too, will be tough to figure out the value. I am fine with the current limitation. I can add a note about it to the docs. I will add paginated tabs as a section to the tabs page.

File: libs/ui/tabs/helm/src/lib/hlm-tabs-paginated-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	template: `
+		<button
+			#previousPaginator
+			type="button"
+			aria-hidden="true"
+			hlmBtn
+			variant="ghost"
+			size="icon"
+			tabindex="-1"
+			[ngClass]="{
```
[the remark this answers is not in the document]
Comment (marcjulian) [reply libs/ui/tabs/helm/src/lib/hlm-tabs-paginated-list.component.ts]: That didn't work correctly. I changed it to 

```html
[class.flex]="_showPaginationControls"
[class.hidden]="!_showPaginationControls"
```

**Comment by goetzrobin** [comment]
Cool that's awesome! Maybe we can make the docs more fun at a later date, but definitely not a blocker! This is great!

**Comment by marcjulian** [comment]
Good idea to improve the docs. Adding all inputs/outputs of a component could also be helpful.

**Comment by goetzrobin** [comment]
@marcjulian is this ready to go?

**Comment by marcjulian** [comment]
@goetzrobin yes is ready to go. I will solve the merge conflicts.

**Pull request merged**